nipype/interfaces/ants/segmentation.py

```diff
@@ -384,6 +387,23 @@ def _list_outputs(self):
                 self._gen_filename('bias_image'))
         return outputs
 
+    def _run_interface(self, runtime, correct_return_codes=(0,)):
+        runtime = super(N4BiasFieldCorrection, self)._run_interface(
+            runtime, correct_return_codes)
+
+        if self.inputs.copy_header:
+            import nibabel as nb
+            refnii = nb.load(self.inputs.input_image)
+            hdr = refnii.header.copy()
+            out_file = self._gen_filename('output_image')
+            nii = nb.load(out_file)
+            hdr.set_data_dtype(nii.header.get_data_dtype())
+            nb.Nifti1Image(nii.get_data(), refnii.affine, hdr).to_filename(
+                out_file)
```

Comment:
Does `N4BiasFieldCorrection` always use NIfTI images, or are we making this assumption? To be more general, this could be:

```Python
if self.inputs.copy_header and runtime.returncode in correct_return_codes:
    import nibabel as nb
    in_img = nb.load(self.inputs.input_image)
    out_file = self._gen_filename('output_image')
    out_img = nb.load(out_file, mmap=False)
    new_img = out_img.__class__(out_img.get_data(), in_img.affine, in_img.header)
    new_img.set_data_dtype(out_img.get_data_dtype())
    new_img.to_filename(out_file)
```

A couple additional notes:

* Should make sure we ran successfully before reading/overwriting the output
* `mmap=False` on `nb.load()` - See nipy/nibabel#492
* No need to copy header (that's done in [image creation](https://github.com/nipy/nibabel/blob/master/nibabel/filebasedimages.py#L192))
* I prefer to do `get/set_data_dtype` on the image itself, rather than the header, even though they're equivalent. Since an image contains a reference to the header and not the other way around, if some image class needs (in a future nibabel version) to modify the data portion of the image during this call, it can. Your call on this one.